File: solutions/erlang/leap/src/leap.erl
-module(leap).

-export([leap_year/1]).

leap_year(Year) ->
    Divisible_by = fun(N) -> Year rem N == 0 end,
    Divisible_by(4) and
        not Divisible_by(100) or
        Divisible_by(400).

File: solutions/c/leap/leap.c
#include "leap.h"

bool leap_year(int year)
{
    return is_divisible_by(year, 4) &&
           (!is_divisible_by(year, 100) ||
            is_divisible_by(year, 400));
}

bool is_divisible_by(int n, int d)
{
    return n % d == 0;
}